examples/dreambooth/train_dreambooth_lora_hidream.py

```diff
@@ -1367,20 +1362,39 @@ def compute_text_embeddings(prompt, text_encoding_pipeline):
 
     vae_config_scaling_factor = vae.config.scaling_factor
     vae_config_shift_factor = vae.config.shift_factor
-    if args.cache_latents:
+
+    # if cache_latents is set to True, we encode images to latents and store them.
+    # Similarly to the pre-encoding in the case of a single instance prompt, if custom prompts are provided
+    # we encode them in advance as well (by iterating over training batches)
+    precompute_latents = args.cache_latents or train_dataset.custom_instance_prompts
+    if precompute_latents:
+        t5_prompt_cache = []
+        llama3_prompt_cache = []
+        pooled_prompt_cache = []
         latents_cache = []
         if args.offload:
             vae = vae.to(accelerator.device)
         for batch in tqdm(train_dataloader, desc="Caching latents"):
             with torch.no_grad():
-                batch["pixel_values"] = batch["pixel_values"].to(
-                    accelerator.device, non_blocking=True, dtype=vae.dtype
-                )
-                latents_cache.append(vae.encode(batch["pixel_values"]).latent_dist)
+                if args.cache_latents:
+                    batch["pixel_values"] = batch["pixel_values"].to(
+                        accelerator.device, non_blocking=True, dtype=vae.dtype
+                    )
+                    latents_cache.append(vae.encode(batch["pixel_values"]).latent_dist)
+                if train_dataset.custom_instance_prompts:
+                    t5_prompt_embeds, llama3_prompt_embeds, pooled_prompt_embeds = compute_text_embeddings(
+                        batch["prompts"], text_encoding_pipeline
+                    )
+                    t5_prompt_cache.append(t5_prompt_embeds)
+                    llama3_prompt_cache.append(llama3_prompt_embeds)
+                    pooled_prompt_cache.append(pooled_prompt_embeds)
 
+            # delete tokenizers and text ecnoders except for llama (tokenizer & te four)
+            # as it's needed for inference with pipeline
         if args.validation_prompt is None:
-            del vae
-            free_memory()
+            del vae, tokenizer_four, text_encoder_four
```

Comment:
Important to note:
https://github.com/huggingface/diffusers/issues/11376#issue-3008144624